docs/docs/sample-scripts/entra/find-obsolete-m365-groups/index.mdx

```diff
@@ -0,0 +1,288 @@
+    if ((m365 status --output text) -eq "Logged out") {
+      m365 login
+    }
+  }
+
+  function Initialize-ExportPath {
+    [CmdletBinding()]
+    param (
+      [Parameter(Mandatory = $false)] [Switch] $KeepOutputPath
+    )
+
+    if (!$KeepOutputPath.IsPresent -or $null -eq $Global:Path) {
+      $Script:Path = Read-Host "Set the path to the folder where you want to export the report data as csv file"
+    }
+    
+    $TestPath = Test-Path -Path  $Script:Path
+    $tStamp = (Get-Date).ToString("yyyyMMddHHmmss")
+    if ($TestPath -ne $true) {
+      New-Item -ItemType directory -Path $Script:Path | Out-Null
+      Write-Host "Will create file in $($Script:Path): M365GroupsReport-{current date}.csv" -ForegroundColor Yellow
+    }
+    else {
+      Write-Host "Following report file will be created in $($Script:Path): 'M365GroupsReport-$($tStamp).csv'."
+      Write-Host "`nAll data will be exported to $($Script:Path): M365GroupsReport-$($tStamp).csv." -ForegroundColor Blue
+      Write-Host "Do not edit this file during the scan." -ForegroundColor Blue
+    }
+    $Script:ReportPath = "$($Script:Path)/M365GroupsReport-$($tStamp).csv"
+  }
+
+  function Get-AllM365Groups {
+    $groups = m365 entra m365group list --includeSiteUrl | ConvertFrom-Json
+    $Script:Groups = $groups | Where-Object { $null -ne $_.siteUrl }
+  }
+
+  function Start-GroupInsightsTests {
+    Write-Host "Checking your $($Script:Groups.Count) groups for activity"
+
+    $Script:Groups | ForEach-Object {
+      $groupInfo = [GroupInfo]::new()
+      $groupInfo.Reference = $_
+      $groupInfo.Membership = @{Owners = 0; Members = 0; Guests = 0 }
+      $groupInfo.TestStatus = "🟢 OK"
+
+      Write-Host "☀︎ $($groupInfo.Reference.displayName)"
+
+      # Tests
+      Test-GroupMembership -Group $groupInfo
+      Test-SharePointActivity -Group $groupInfo
+      Test-ConversationActivity -Group $groupInfo
+
+      # Report
+      New-Report -Group $groupInfo
+    }
+    
+    #Give feedback to user
+    Write-Host "`n-------------------------------------------------------------------"
+    Write-Host "`SUMMARY" -ForegroundColor DarkGreen
+    Write-Host "`-------------------------------------------------------------------"
+    Write-Host "`n👉 Found $($Global:ObsoleteGroups.Count) group$($Global:ObsoleteGroups.Count -gt 1 ? 's' : '') with possibly low activity."
+    Write-Host "`   Please review the report: " -NoNewline
+    Write-Host "$($Script:ReportPath)" -ForegroundColor DarkBlue
+  }
+
+  function Test-GroupMembership {
+    [CmdletBinding()]
+    param (
+      [Parameter(Mandatory = $true)] [GroupInfo] $Group
+    )
+
+    $users = m365 entra m365group user list --groupId $Group.Reference.id | ConvertFrom-Json
+    $owners = $users | Where-Object { $_.userType -eq "Owner" }
+    $members = $users | Where-Object { $_.userType -eq "Member" }
+    $guests = $users | Where-Object { $_.userType -eq "Guest" }
```

Comment:
The property `userType` is going to be removed in the next major release (https://github.com/pnp/cli-microsoft365/issues/5558). It was never our intention to create this property. People should use `roles` to check if the user is a member or an owner.
Note that `Guest` is not a role. If you want to know if a user is a guest, I think your best shot will be to run `entra user list` and check there if the user is a guest.

Reply:
Thanks for this notice, although it's a bit confusing: 
in the documented list of examples for `entra m365group user list` (https://pnp.github.io/cli-microsoft365/cmd/entra/m365group/m365group-user-list#examples), you find this: `m365 entra m365group user list --groupDisplayName Developers --filter "userType eq 'Guest'"`. In this case, we should remove the example.

As the `Guest` role is now deprecated, is it correct that to be a `Guest` user, we still can check whether a user is neither `Owner` nor `Member`?

Reply:
Good point @tmaestrini, it seems like we overlooked removing the example.
You can check if a user is member of owner indeed, but we'll have to check the `roles` property for that. If you want to know if a user is a guest, you'll have to check for that user in Microsoft Entra ID (`entra user get`) for example.
But for me, it's not required to do this check. If you're fine implementing it, go ahead, but it's not required 😊

Reply:
I've changed the implementation to distinguish between owners, members and guest according to the `role` property.